docs/static/config-details.asciidoc

```diff
@@ -87,6 +90,35 @@ Keep the overall memory requirements in mind when you allocate memory.
 By default, a JVM's off-heap direct memory limit is the same as the heap size. Check out <<plugins-inputs-beats-memory,beats input memory usage>>.
 Consider setting `-XX:MaxDirectMemorySize` to half of the heap size.
 
+Internal JVM data structures, thread stacks, memory mapped files, direct memory for input/output (IO) operations, and the Java heap used to execute Logstash all use JVM memory. 
+Keep the memory requirements for these processes in mind as you calculate your ideal memory allocation.
+In addition to Java heap and direct memory, you must account for native memory as part of the overall memory consumption for {ls} processes.
+
+Java heap can be controlled by Xmx and Xms settings. 
+Direct memory space is controlled by the `-XX:MaxDirectMemorySize` setting. 
+Total JVM memory allocation must be estimated and is controlled indirectly using Java heap and direct memory settings.
+
+As you make your capacity calculations, keep in mind that the JVM can't consume more than 70% of the host memory available.
+
+For a {ls} instance with persistent queue (PQ) enabled on multiple pipelines, we could
+estimate memory consumption using:
+
+[source,text]
+-----
+pipelines number * (workers * stack size + 2 * PQ page size) + direct memory + Java heap
+-----
```

Comment:
maybe add a note here explaining one of the biggest memory hogs, "2* PQ page size", with something like "a PQ may page into memory up to two queue pages".

Reply:
Fixed by b352be45ace4811aa55057187abe3a572f933e4f